src/store/traits.rs

```diff
@@ -90,6 +95,31 @@ pub trait MapEntry: std::fmt::Debug + Clone + Send + Sync + 'static {
     fn outboard(&self) -> impl Future<Output = io::Result<impl Outboard>> + Send;
     /// A future that resolves to a reader that can be used to read the data
     fn data_reader(&self) -> impl Future<Output = io::Result<impl AsyncSliceReader>> + Send;
+
+    /// Encodes data and outboard into a [`AsyncStreamWriter`].
+    ///
+    /// Data and outboard parts will be interleaved.
+    ///
+    /// `offset` is the byte offset in the blob to start the stream from. It will be rounded down to
```

Comment:
Are you sure this will round down to the next chunk group? The ranges are given in chunks, not chunk groups, so it should round down to the next chunk. So e.g. 1026 would round down to 1024.

In addition: I am sure both sides are compatible with each other, but I am not sure if they are compatible with bao without chunk groups, as specified in WGPS. Do we care?

What bao-tree will do is the following: It will round down to individual chunks, but only where needed. So e.g. if you ask for 31*1024..65*1024, it will send the chunk from 32*1024..64*1024 as a single chunk of data without any intermediate hashes. If you want to be compatible with bao without chunk groups you would have to send this as 16 individual 1024 byte chunks with intermediate hash pairs.

Reply:
> In addition: I am sure both sides are compatible with each other, but I am not sure if they are compatible with bao without chunk groups, as specified in WGPS. Do we care?

[WGPS allows payload transformations](https://willowprotocol.org/specs/sync/index.html#sync_payloads_transform). So it's just the iroh flavor of WGPS payload transfer.

> Are you sure this will round down to the next chunk group? The ranges are given in chunks, not chunk groups, so it should round down to the next chunk. So e.g. 1026 would round down to 1024.

I need to have a proper look. I've mostly shuffled around the code, I'm just the messenger :)